Add FA weights to CIP-0104 spec (#238)
global-synchronizer-foundation/cips: cip-0104/cip-0104.md

Reviewer: What about adjusting the example with one non-default weight to create certainty?

Author: @meiersi-da Could you double check my example 4, please? If it's fine, please tag @hythloda that all comments are addressed.

Reviewer: yes, that is correct. Thanks for adding it.

Reviewer: @hythloda : all comments are addressed. I can't resolve my own comment, so I'll have to leave that to you.

Reviewer: ```suggestion
The computations in steps 1-5 are performed using integer arithmetic for efficiency and determinism reasons.
```

Reviewer: ```suggestion
For confirmation requests whose traffic size is below 100 MB, they can be performed with 64-bit signed integers without risk of overflow.
The computation in Step 6 is performed using Daml's `Decimal` arithmetic as `floor (intToDecimal per_app_traffic_weight * activityWeight)`.
```

   ```

   using integer arithmetic, and attribute `per_app_traffic_weight` to every app confirmer of the envelope.

> 6. For every featured app involved, multiply the `per_app_traffic_weight` by the `activity_weight` set on the `FeaturedAppRight`.

Reviewer: ```suggestion
6. For every featured app involved, multiply the `per_app_traffic_weight` by the `activityWeight` set on the `FeaturedAppRight`.
```

1. Traffic summaries can be read by clients as soon as the corresponding message has been sequenced, and not only once the confirmation request has been completed, which may be up to 30s later.
2. The `/v2/events` stream is not held up by a potentially delayed app activity record computation.

#### Featured App Weights

The CIP proposes to add the ability to adjust the app rewards that a featured app receives by setting a weight on the `FeaturedAppRight` contract. The weight is applied in the [activity record computation](#activity-record-computation-details).
Concretely:

> 1. Add a field `activity_weight` to `FeaturedAppRight`.

Reviewer: ```suggestion
1. Add a field `activityWeight` to `FeaturedAppRight`.
```

> 2. Adjust the `DsoRules_GrantFeaturedAppRight` and `DsoRules_RevokeFeaturedAppRight` governance choices to optionally take `activity_weight`, with the default being `1.0`.

Reviewer: ```suggestion
2. Adjust the `DsoRules_GrantFeaturedAppRight` governance choice to optionally take `activityWeight`, with the default being `1.0`.
```

> 3. Add a new `DsoRules_UpdateFeaturedAppRight` governance choice to adjust the `activity_weight`.

Reviewer: ```suggestion
3. Add a new `DsoRules_UpdateFeaturedAppRight` governance choice to adjust the `activityWeight`.
```

> 4. Make the adjustments to the SV UIs and backing APIs to allow SV governance to adjust the `activity_weight` on `FeaturedAppRight`

Reviewer: ```suggestion
4. Adjust the SV UIs and backing APIs to allow SV governance to change the `activityWeight` on `FeaturedAppRight`
```

### Free Protocol-Conformant Confirmation Responses

> The Canton Protocol is changed so that the confirmation responses expected to be sent by a confirming participant node become de-facto free.

Reviewer: Missing later below: adjust the changelog. Also have a look at the header. I believe there's some changelog as well.

Author: I adjusted the changelog at the bottom, I saw nothing at the top.
